OSOE-1109: Add simple EN 16931 XML eInvoice validator

wAsnk opened the pull request:
[OSOE-1109](https://lombiq.atlassian.net/browse/OSOE-1109)

[OSOE-1109]: https://lombiq.atlassian.net/browse/OSOE-1109?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

wAsnk committed:
Add new submodule

wAsnk committed:
Merge remote-tracking branch 'origin/dev' into issue/OSOE-1109

wAsnk committed:
Merge remote-tracking branch 'origin/dev' into issue/OSOE-1109

wAsnk committed:
Submodule src/Libraries/Lombiq.EInvoiceValidator updated

Submodule src/Libraries/Lombiq.EInvoiceValidator:
    > e3bf2a8 - Better naming
    > 1cbdde5 - Docs
    > dbc4637 - Use dotnet run instead
    > 93f2c1d - Pack script file
    > e262c3e - Not needed
    > 982547e - Using npm targets
    > 14dc14c - Update nuget build conditions
    > 3c0fca5 - Usings
    > ab026b5 - Add unit test
    > be445de - Rename
    > 5fa84de - More explanation
    > d5b6c7b - Add simple sample
    > aec5de6 - Add tostring override
    > eb70fe8 - More refactoring because of less detailed time measurements
    > eefa943 - Add memory cache directly here, because it is a dependency
    > 43e0ea4 - Add comments
    > 2c44ec8 - Format code
    > cc67f51 - Refactor benchmark
    > 6a86988 - Delete time monitoring from main library
    > 45bbe36 - Add benchmark console project
    > 2bdd373 - Refactor the basics and create the main base version with future proof logic
    > 7c1830b - Update gitignore
    > 8d6d023 - Generate files
    > b37caec - Adding validation file download and generator for EN 16931
    > f30afc9 - Adding basics of schematron validation

wAsnk committed:
Add new library projects

wAsnk committed:
Submodule src/Libraries/Lombiq.EInvoiceValidator updated

Submodule src/Libraries/Lombiq.EInvoiceValidator:
    > 9a3a1a1 - Add readme
    > 3d614d8 - Small refactoring and docs

wAsnk committed:
Submodule src/Libraries/Lombiq.EInvoiceValidator updated

Submodule src/Libraries/Lombiq.EInvoiceValidator:
    > 43255bc - Using American English

wAsnk committed:
Add exceptions

wAsnk committed:
Adjust skips

wAsnk committed:
Submodule src/Libraries/Lombiq.EInvoiceValidator updated

Submodule src/Libraries/Lombiq.EInvoiceValidator:
    > 116d2b0 - Codespell violation fix

wAsnk committed:
Adjusting skip

wAsnk committed:
Submodule src/Libraries/Lombiq.EInvoiceValidator updated

Submodule src/Libraries/Lombiq.EInvoiceValidator:
    > 20cfcf5 - Adding additional info about validation artifacts
    > 298eea9 - Trying to fix PS code analyzer errors
    > db3b1bc - Final touches
    > 49b208d - Include node modules

wAsnk committed:
Submodule src/Libraries/Lombiq.EInvoiceValidator updated

Submodule src/Libraries/Lombiq.EInvoiceValidator:
    > 42caace - To new line

wAsnk committed:
Submodule src/Libraries/Lombiq.EInvoiceValidator updated

Submodule src/Libraries/Lombiq.EInvoiceValidator:
    > 9d51952 - New lines

wAsnk committed:
Submodule src/Libraries/Lombiq.EInvoiceValidator updated

Submodule src/Libraries/Lombiq.EInvoiceValidator:
    > da9da9c - .net consolidation, match with other packages

wAsnk committed:
Submodule src/Libraries/Lombiq.EInvoiceValidator updated

Submodule src/Libraries/Lombiq.EInvoiceValidator:
    > 2e0c681 - We don't need to use nuget packages
    > 5b23910 - Not needed
    > 272a41b - Reference sample files
    > 194c73f - Adding include for source references and docs
    > dae8821 - Reordering
    > 872fe5a - Deleting double files

wAsnk committed:
Submodule src/Libraries/Lombiq.EInvoiceValidator updated

Submodule src/Libraries/Lombiq.EInvoiceValidator:
    > ef14db0 - More docs
    > 78f3bee - Fix after moving sample files
    > c8a69e2 - Cleanup

wAsnk committed:
Submodule src/Libraries/Lombiq.EInvoiceValidator updated

Submodule src/Libraries/Lombiq.EInvoiceValidator:
    > 56d98d3 - Fix readme

Piedone commented on src/Libraries/Lombiq.EInvoiceValidator:
Please publish an alpha already, and add it to the NuGetTest solution.

Piedone committed:
Submodule src/Libraries/Lombiq.EInvoiceValidator updated

Submodule src/Libraries/Lombiq.EInvoiceValidator:
    > 2acc8d3 - Code styling
    > cc38037 - Docs formatting

wAsnk committed:
Submodule src/Libraries/Lombiq.EInvoiceValidator updated

Submodule src/Libraries/Lombiq.EInvoiceValidator:
    > 8750dce - Lower case
    > 316651e - Add docs
    > 1df0da4 - Add files as member data
    > aa10a9c - Rename
    > d7c5692 - Nullable changes
    > e03e7fa - Filtering node modules
    > 4eedafc - Project wide nullable
    > 7aa3234 - Moving to existing services
    > 84e0a5c - Remove empty readme
    > 1a7103e - Update Readme.md
    > 59347d6 - Add more info
    > a12a9f9 - Move to services
    > 9233fbf - Adjust docs
    > 7f7782a - Create folders when needed
    > cff01de - Add editorconfig
    > 447c3c7 - Delete
    > f72c180 - Add references to projects
    > d4ab561 - Add license
    > 9ef9160 - Add readme
    > 38dd57a - eInvoice
    > 2acc8d3 - Code styling
    > cc38037 - Docs formatting

wAsnk committed:
Add linting

wAsnk committed:
Merge remote-tracking branch 'origin/issue/OSOE-1109' into issue/OSOE-1109

wAsnk committed:
Submodule src/Libraries/Lombiq.EInvoiceValidator updated

Submodule src/Libraries/Lombiq.EInvoiceValidator:
    > 1184515 - Typo fix

wAsnk committed:
Missing comma

wAsnk committed:
Submodule src/Libraries/Lombiq.EInvoiceValidator updated

Submodule src/Libraries/Lombiq.EInvoiceValidator:
    > ef516bf - Fixing suggestion

wAsnk committed:
Submodule src/Libraries/Lombiq.EInvoiceValidator updated

Submodule src/Libraries/Lombiq.EInvoiceValidator:
    > c5d67ad - Include all except not needed.
    > 61e9b27 - Name change

wAsnk committed:
Add eInvoice testing

Piedone committed:
Merge branch 'dev' into issue/OSOE-1109

Piedone commented on NuGetTest/test/Lombiq.OSOCE.NuGet.Tests/Lombiq.OSOCE.NuGet.Tests.csproj:
```diff
@@ -16,4 +16,12 @@
   </ItemGroup>
 
+  <ItemGroup>
+    <ProjectReference Include="..\..\src\Lombiq.OSOCE.NuGet.Web\Lombiq.OSOCE.NuGet.Web.csproj" />
```
The non-UI test project shouldn't have to reference the web project. Why is this needed?

Piedone commented on NuGetTest/test/Lombiq.OSOCE.NuGet.Tests/Lombiq.OSOCE.NuGet.Tests.csproj:
```diff
@@ -16,4 +16,12 @@
+
+    <None Include="SampleInvoices\**\*.xml">
+      <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
+    </None>
```
These files shouldn't be copied here but rather used from the project.

wAsnk replied:
Do you want to NuGet release the sample project and include those files in there?

wAsnk replied:
Did that

Piedone replied:
I was thinking about referencing the files in the Samples project directly from the test project, but this is OK as well.

Piedone commented on .github/workflows/build-and-test.yml:
```diff
@@ -68,7 +68,8 @@ jobs:
         src/Modules/Lombiq.UIKit/Lombiq.UIKit,
-        src/Modules/Lombiq.Walkthroughs/Lombiq.Walkthroughs
+        src/Modules/Lombiq.Walkthroughs/Lombiq.Walkthroughs,
+        src/Libraries/Lombiq.EInvoiceValidator/Lombiq.EInvoiceValidator
```
This doesn't lint the JS files in the project, since they aren't in the wwwroot, what this simplified config requires. Either move them to there, though that would be hackish, all use the full config I linked, in a workflow in the project: https://github.com/Lombiq/NodeJs-Extensions?tab=readme-ov-file#full-configuration

Piedone replied:
While I was thinking about a workflow in the Validator project and keeping this workflow here intact, changing this workflow is even better.

wAsnk committed:
Submodule src/Libraries/Lombiq.EInvoiceValidator updated

Submodule src/Libraries/Lombiq.EInvoiceValidator:
    > 3f3906f - Add to solution to release to NuGet
    > 2e352a7 - Update readme
    > 0780e52 - Add NuGet properties
    > 1638152 - Update readme
    > f26c989 - Update readme

wAsnk committed:
Move reference

wAsnk committed:
Convert to full configuration

wAsnk committed:
Merge remote-tracking branch 'origin/issue/OSOE-1109' into issue/OSOE-1109

# Conflicts:
#	NuGetTest/test/Lombiq.OSOCE.NuGet.Tests/Lombiq.OSOCE.NuGet.Tests.csproj

wAsnk committed:
Commas

wAsnk committed:
Trying base

wAsnk committed:
Submodule src/Libraries/Lombiq.EInvoiceValidator updated

Submodule src/Libraries/Lombiq.EInvoiceValidator:
    > fb12d0a - Little fixes

wAsnk committed:
Use new sample nuget

Submodule src/Libraries/Lombiq.EInvoiceValidator updated

Submodule src/Libraries/Lombiq.EInvoiceValidator:
    > d69691d - Add shield

Piedone committed:
Submodule src/Libraries/Lombiq.EInvoiceValidator updated

Submodule src/Libraries/Lombiq.EInvoiceValidator:
    > 058d8ff - Formatting, clarifying Node.js

Piedone committed:
Merge branch 'dev' into issue/OSOE-1109

# Conflicts:
#	Lombiq.OSOCE.sln
